src/route-loader.ts

```diff
@@ -0,0 +1,74 @@
+import { Origin } from 'aurelia-metadata';
+import { relativeToFile } from 'aurelia-path';
+import { NavigationInstruction, RouteConfig, RouteLoader, Router, ViewPortComponent } from 'aurelia-router';
+import { CompositionEngine, customElement, inlineView, useView } from 'aurelia-templating';
+import { RouterViewLocator } from './router-view';
+
+class EmptyClass { }
+inlineView('<template></template>')(EmptyClass);
+
+export class TemplatingRouteLoader extends RouteLoader {
+
+  static inject = [CompositionEngine];
+
+  compositionEngine: CompositionEngine;
+
+  constructor(
+    compositionEngine: CompositionEngine
+  ) {
+    super();
+    this.compositionEngine = compositionEngine;
+  }
+
+  loadRoute(router: Router, config: RouteConfig, _navInstruction: NavigationInstruction): Promise<ViewPortComponent> {
```

Comment:
where did _navInstruction come from?

Reply:
I was only completing the signature, since the router does pass a 3rd parameter of type `NavigationInstruction`